courseware/api.py

```diff
@@ -1,36 +1,50 @@
 """Courseware API functions"""
-from datetime import datetime, timedelta
-from urllib.parse import urljoin
-import pytz
+from datetime import timedelta
+from urllib.parse import urljoin, urlencode, urlparse, parse_qs
 import requests
 from rest_framework import status
 
 from django.conf import settings
+from django.db import transaction
+from django.shortcuts import reverse
 from oauth2_provider.models import Application, AccessToken
 from oauthlib.common import generate_token
 
-from courseware.models import CoursewareUser
+from authentication import api as auth_api
+from courseware.exceptions import OpenEdXOAuth2Error
+from courseware.models import CoursewareUser, OpenEdxApiAuth
 from courseware.constants import PLATFORM_EDX
+from mitxpro.utils import now_in_utc
 
 
 OPENEDX_REGISTER_USER_PATH = "/user_api/v1/account/registration/"
 OPENEDX_REQUEST_DEFAULTS = dict(country="US", honor_code=True)
 
+OPENEDX_SOCIAL_LOGIN_XPRO_PATH = "/auth/login/mitxpro-oauth2/?auth_entry=login"
```

Comment:
Would be nice to have a settings var for `mitxpro-oauth2` at this point

Reply:
Is there a reason for this to change? It's set by the installed backend in edx and very likely won't change because we'd have to reissue auths for every user somehow.